### PR TITLE
INT: Replace Max Hop Count and Total Hop Count with Remaining Hop Count

### PR DESCRIPTION
**Pull request by mickeyspiegel**

### CODE REVIEW

File: telemetry/specs/INT.mdk [hunk 1]
```diff
@@ -687,17 +687,17 @@ order of bits set in the instruction bitmap.
...
-Instruction Bitmap.
+    - Ver, Rep, C, Instruction Count, Remaining Hop Count, and Instruction
+      Bitmap.
     - INT Source must set all reserved bits to zero.
   - Intermediate devices can set the following fields:
-    - C, E, Total Hop Count
+    - C, E, Remaining Hop Count
 * In an INT packet, the length (in bytes) of the INT metadata stack must always
-be (Instruction Count \* 4 \* Total Hop Count).
+be a multiple of (Instruction Count \* 4).
```
Comment (jklr) [inline telemetry/specs/INT.mdk]: Shall we clarify how INT Sink can derive the total metadata bytes from the 'length' field in shim header? Every encap -- INT over TCP/UDP, vxlan-gpe, and geneve -- has shim or option header, which includes the total word length. The sink can compute 
   metadata stack size = 'length' * 4 - [total fixed header sizes]
The total fixed header sizes must be known for each encap.

**Review by jklr — APPROVED** [review]

File: telemetry/specs/INT.mdk [hunk 2]
```diff
@@ -878,7 +877,7 @@ INT Metadata Header and Metadata Stack:
     0                   1                   2                   3
     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-   |  Ver  |Rep|C|E|  R  | InsCnt=2|  MaxHopCnt=16 | TotalHopCnt=3 |
+   |  Ver  |Rep|C|E|  R  | InsCnt=2|   Reserved    |RemainingHopC=5|
```
Comment (mhira1) [inline telemetry/specs/INT.mdk]: Could you please also add M bit here? I missed this example in my commit.

File: telemetry/specs/INT.mdk [hunk 3]
```diff
@@ -998,7 +997,7 @@ INT Metadata Header and Metadata Stack:
     0                   1                   2                   3
     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-   |  Ver  |Rep|C|E|  R  | InsCnt=1|  MaxHopCnt=16 | TotalHopCnt=3 |
+   |  Ver  |Rep|C|E|  R  | InsCnt=1|   Reserved    |RemainingHopC=5|
```
Comment (mhira1) [inline telemetry/specs/INT.mdk]: Could you please also add M bit here? I missed this example in my commit.

File: telemetry/specs/INT.mdk [hunk 4]
```diff
@@ -734,14 +734,14 @@ both metadata (switch id and queue occupancy).
...
 intermediate switches must process) is 1, and that the type value for the
 destination INT Header is 2.
+* The maximum number of hops (network diameter) is 8.
 * The INT metadata header uses the following field values in the metadata header,
 unless specified otherwise in each example.
   - Ver = 0
   - Rep = 0 (No replication)
   - C = 0
   - E = 0 (Max Hop Count not exceeded)
   - Instruction Count = 2 (for switch id & queue occupancy)
```
Comment (mhira1) [inline telemetry/specs/INT.mdk:775]: May be say here that the INT source starts with a remaining hop count = 8

**Review by mhira1 — APPROVED** [review]
Looks good, minor comments inline

File: telemetry/specs/INT.mdk (hunk 4 above)
Comment (mhira1) [reply telemetry/specs/INT.mdk:775]: Sorry, noticed that you do say maximum number of hops (network diameter) is 8. Hence INT source starts with a remaining hop count of 8.

File: telemetry/specs/INT.mdk (hunk 1 above)
Comment (mhira1) [reply telemetry/specs/INT.mdk]: Agree, let's add some clarification on how INT sink can derive total metadata length.

File: telemetry/specs/INT.mdk [hunk 5]
```diff
@@ -575,7 +575,7 @@ INT Metadata Header and Metadata Stack:
     0                   1                   2                   3
     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5     6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-   |  Ver  |Rep|C|E|M|R R| Ins Cnt |  Max Hop Cnt  | Total Hop Cnt |
+   |  Ver  |Rep|C|E|M|R R| Ins Cnt |   Reserved    |RemainingHopCnt|
```
Comment (mhira1) [inline telemetry/specs/INT.mdk]: Wouldn't it be better to make reserved bits contiguous? i.e. Shift Instruction count to the right, adjacent to RemainingHopCnt?

File: telemetry/specs/INT.mdk (hunk 1 above)
Comment (mickeyspiegel) [reply telemetry/specs/INT.mdk]: Done

File: telemetry/specs/INT.mdk (hunk 2 above)
Comment (mickeyspiegel) [reply telemetry/specs/INT.mdk]: Done

File: telemetry/specs/INT.mdk (hunk 3 above)
Comment (mickeyspiegel) [reply telemetry/specs/INT.mdk]: Done

File: telemetry/specs/INT.mdk (hunk 5 above)
Comment (mickeyspiegel) [reply telemetry/specs/INT.mdk]: Done

**Review by mhira1 — APPROVED** [review]

**Pull request merged**